lib/chef/knife/s3_source.rb

```diff
@@ -20,11 +20,11 @@ def bucket_obj
       end
 
       def bucket
-        URI(@url).host
+        URI(@url).path.split("/")[1]
```

Comment:
@eherot , I tried with `s3://` style URL too eg. `s3://provisioning.bucket.com/chef/encrypted_data_bag_secret`. In this case if we go with previous code, then `bucket` value comes out to be `provisioning.bucket.com`, which breaks as the `bucket` should be `chef`. Can you suggest me a valid `s3://` URL format which is supported by previous code? We'll modify the code accordingly.